### PR TITLE
Lower MSRV to `1.57`

### PR DESCRIPTION
**Pull request by devtomio**
## What does this PR address?

Lowers the _**M**inimum **S**upported **R**ust **V**ersion_ of Nixpacks from [`1.58.1`](https://blog.rust-lang.org/2022/01/20/Rust-1.58.1.html) to [`1.57`](https://blog.rust-lang.org/2021/12/02/Rust-1.57.0.html).

## Before submitting:

- [x] Make sure to follow [GitHub's guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on creating PR.
- [x] Did you read through [contribution guidelines](https://github.com/railwayapp/nixpacks/blob/main/CONTRIBUTING.md)?
- [x] Did your changes require updates to the documentation? Have you updated those accordingly?
- [ ] Did you write tests to cover your changes? Did it pass locally when running `cargo test`?
- [x] Have you run `cargo fmt`, `cargo check` and `cargo clippy` locally to ensure that CI passes?

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -22,13 +23,13 @@ path = "src/lib.rs"
...
 anyhow = "1.0.58"
-clap = { version = "3.2.14", features = ["std"], default-features = false }
+clap = { version = "3.2.16", features = ["std"], default-features = false }
 colored = "2.0.0"
 globset = { default-features = false, version = "0.4.9" }
 walkdir = "2.3.2"
 indoc = "1.0.6"
 regex = "1.6.0"
-serde = { version = "1.0.140", default-features = false}
+serde = { version = "1.0.140", default-features = false }
```
Comment (Milo123459) [inline Cargo.toml:32]: cargo fmt should really format cargo.toml

File: Cargo.toml [hunk 2]
```diff
@@ -42,5 +43,9 @@ box_drawing = "0.1.2"
 textwrap = { default-features = false, version = "0.15.0" }
 cargo_toml = "0.11.5"
 
+# Why does cargo update this dependency to the latest version
+# `clap_lex` uses 6.0 (https://github.com/clap-rs/clap/blob/master/clap_lex/Cargo.toml#L41)
+os_str_bytes = { default-features = false, version = "=6.0", features = ["raw_os_str"] }
```
Comment (Milo123459) [inline Cargo.toml:48]: Do we need to do this? Is there no workaround we can do?

Comment (devtomio) [reply Cargo.toml:48]: There is a workaround. By ignoring the Cargo lockfile (`Cargo.lock`) from the repository. 

- https://github.com/rust-lang/cargo/issues/315
- https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html#:~:text=If%20you%E2%80%99re%20building%20a%20non%2Dend%20product%2C%20such%20as%20a%20rust%20library%20that%20other%20rust%20packages%20will%20depend%20on%2C%20put%20Cargo.lock%20in%20your%20.gitignore.

Comment (Milo123459) [reply Cargo.toml:48]: This project shouldn't ignore Cargo.lock. Does this dependency matter so much that it would increase MSRV?

Comment (devtomio) [reply Cargo.toml:48]: It would increase it to `1.57.0`.

Comment (Milo123459) [reply Cargo.toml:48]: That's fine IMHO.

**Review by Milo123459 — APPROVED** [review]

**Review by coffee-cup — COMMENTED** [review]
What is the reasoning for lowering the MSRV?

**Comment by devtomio** [comment]
> What is the reasoning for lowering the MSRV?

Mainly compatibility with lower Rust versions.

**Pull request merged**